staging/vhost-device-can/src/can.rs

```diff
@@ -0,0 +1,438 @@
+// CAN backend device
+//
+// Copyright 2023-2024 VIRTUAL OPEN SYSTEMS SAS. All Rights Reserved.
+//          Timos Ampelikiotis <t.ampelikiotis@virtualopensystems.com>
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use log::{error, info, trace, warn};
+use std::sync::{Arc, RwLock};
+
+use thiserror::Error as ThisError;
+use vm_memory::{ByteValued, Le16};
+
+extern crate socketcan;
+use socketcan::{
+    CanAnyFrame, CanDataFrame, CanFdFrame, CanFdSocket, EmbeddedFrame, ExtendedId, Frame, Id,
+    Socket, StandardId,
+};
+
+use std::thread::{spawn, JoinHandle};
+use vmm_sys_util::eventfd::{EventFd, EFD_NONBLOCK};
+
+extern crate queues;
+use queues::*;
+
+use crate::vhu_can::VirtioCanFrame;
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Copy, Clone, Debug, PartialEq, ThisError)]
+/// Errors related to low level can helpers
+pub(crate) enum Error {
+    #[error("Can open socket operation failed")]
+    SocketOpen,
+    #[error("Can write socket operation failed")]
+    SocketWrite,
+    #[error("Can read socket operation failed")]
+    SocketRead,
+    #[error("Pop can element operation failed")]
+    PopFailed,
+    #[error("Queue is empty")]
+    QueueEmpty,
+    #[error("Creating Eventfd for CAN events failed")]
+    EventFdFailed,
+    #[error("Push can element operation failed")]
+    PushFailed,
+    #[error("No output interface available")]
+    NoOutputInterface,
+}
+
+/* CAN flags to determine type of CAN Id */
```

Comment:
ditto

Reply:
and also others in this file

Reply:
If you prefer, we can also move these define in a separate file, like virtio_bindings.rs

Not a strong opinion, just an idea.

Reply:
Thanks! That's a good idea. I have placed all those defines in a seperate file called `virtio_bindins.rs`.
(The comments have been also updated to use `//`)